consul/consul.go

```diff
@@ -323,9 +323,13 @@ func (c *Consul) marathonToConsulChecks(task *apps.Task, healthChecks []apps.Hea
 		}
 
 		switch check.Protocol {
-		case "HTTP", "HTTPS":
+		case "HTTP", "HTTPS", "MESOS_HTTP", "MESOS_HTTPS":
 			if parsedURL, err := url.ParseRequestURI(check.Path); err == nil {
-				parsedURL.Scheme = strings.ToLower(check.Protocol)
+				if check.Protocol == "HTTP" || check.Protocol == "MESOS_HTTP" {
```

Comment:
What test would you like to see? I added one [here](https://github.com/allegro/marathon-consul/pull/157/files/0b2daba4b1ab7a3e0cf41131d52198b6e7a6795e#diff-eccf782d4b592a6a09965a14b71f5e08R832)